instruments/qubitekk/mc1.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides support for the Qubitekk MC1 Motor Controller.
+
+MC1 Class originally contributed by Catherine Holloway.
+"""
+
+# IMPORTS #####################################################################
+from __future__ import absolute_import
+from enum import Enum
+from numpy import arange
+
+from instruments.abstract_instruments import Instrument
+# CLASSES #####################################################################
+
+
+class MC1(Instrument):
+    """
+    The MC1 is a controller for the qubitekk motor controller. Used with a
+    linear actuator to perform a HOM dip.
+    """
+    def __init__(self, filelike, increment=1, upper_limit=300,
+                 lower_limit=-300):
+        super(MC1, self).__init__(filelike)
+        self.terminator = "\r"
+        self.increment = increment
+        self.lower_limit = lower_limit
+        self.upper_limit = upper_limit
+        self._firmware = None
+        self._controller = None
+
+    # ENUMS #
+
+    class MotorType(Enum):
+        """
+        Enum for the motor types for the MC1
+        """
+        radio = "Radio"
+        relay = "Relay"
+
+    # PROPERTIES #
+
+    @property
+    def setting(self):
+        """
+        Get the current output setting
+        :return: int, representing the currently active channel
+        """
+        response = self.query("OUTP?")
+        return int(response)
+
+    @setting.setter
+    def setting(self, new_val):
+        """
+        Set the current output setting
+        :param new_val: the output channel number, either 0 or 1
+        :type new_val: either int or boolean
+        """
+        if new_val == 0:
+            self.sendcmd(":OUTP 0")
+        elif new_val == 1:
+            self.sendcmd(":OUTP 1")
+        else:
+            raise ValueError("Setting not recognized")
+
+    @property
+    def position(self):
```

Comment:
The `int_property` factory can be used here with the `readonly` flag set to `True`